src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2654,6 +2657,7 @@ private void handleCopyPart(HttpServletRequest request,
                         his.hash().asBytes());
             } else {
                 Payload payload = Payloads.newInputStreamPayload(is);
+                // TODO: Content-MD5
```

Comment:
Running s3fs tests I found that this middlware cannot handle UploadPartCopy due to the emulation that S3Proxy does.  It cannot get the correct MD5 of a range of data to satisfy the underlying `RequiredContentMD5BlobStore`.